src/s2_cells/mod.rs

```diff
@@ -50,3 +63,157 @@ impl S2Meta {
+pub struct S2CellId(pub CellID);
+
+impl std::fmt::Display for S2CellId {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        write!(f, "{}", self.0.to_token())
+    }
+}
+
+impl PointCloud for S2Cells {
+    type Id = S2CellId;
+    type PointsIter = FilteredPointsIterator;
+
+    fn nodes_in_location(&self, query: &PointQuery) -> Vec<Self::Id> {
+        match &query.location {
+            PointLocation::AllPoints() => self.cells.keys().cloned().map(S2CellId).collect(),
+            PointLocation::Aabb(aabb) => {
+                self.cells_in_obb(&Obb::from(aabb), query.global_from_local.as_ref())
+            }
+            PointLocation::Obb(obb) => self.cells_in_obb(&obb, query.global_from_local.as_ref()),
+            PointLocation::Frustum(mat) => {
+                let world_from_clip = mat.inverse_transform().unwrap();
+                let points = CUBE_CORNERS
+                    .iter()
+                    .map(|p| Point3::from_homogeneous(world_from_clip * p));
+                self.cells_in_convex_hull(points)
+            }
+            PointLocation::OrientedBeam(_) => unimplemented!(),
+        }
+    }
+
+    fn encoding_for_node(&self, _: Self::Id) -> Encoding {
+        Encoding::Plain
+    }
+
+    fn points_in_node<'a>(
+        &'a self,
+        query: &PointQuery,
+        node_id: Self::Id,
+    ) -> Result<Self::PointsIter> {
+        let culling = query.get_point_culling();
+        let num_points = self.meta.cells[&node_id.0].num_points as usize;
+        let node_iterator = NodeIterator::from_data_provider(
+            &*self.data_provider,
+            self.encoding_for_node(node_id),
+            &node_id,
+            num_points,
+        )?;
+        Ok(FilteredPointsIterator {
+            culling,
+            node_iterator,
+        })
+    }
+}
+
+impl S2Cells {
+    /// Wrapper arround cells_in_convex_hull for Obbs
+    fn cells_in_obb(
+        &self,
+        obb: &Obb<f64>,
+        global_from_local: Option<&Isometry3<f64>>,
+    ) -> Vec<S2CellId> {
+        let obb = match global_from_local {
+            Some(isometry) => Cow::Owned(Obb::new(
+                isometry * &obb.isometry_inv.inverse(),
+                obb.half_extent,
+            )),
+            None => Cow::Borrowed(obb),
+        };
+        let points = obb.corners;
+        self.cells_in_convex_hull(points.iter().cloned())
+    }
+
+    /// Returns all cells that intersect the convex hull of the given points
+    fn cells_in_convex_hull<IterPoint>(&self, points: IterPoint) -> Vec<S2CellId>
+    where
+        IterPoint: IntoIterator<Item = Point3<f64>>,
+    {
+        // We could choose either a covering rect or a covering cap as a convex hull
+        let point_cells = points
+            .into_iter()
+            .map(|p| CellID::from(S2Point::from_coords(p.x, p.y, p.z)))
+            .collect();
+        let mut cell_union = CellUnion(point_cells);
+        cell_union.normalize();
```

Comment:
Referring to the comment above, is this what we want? We will potentially have cells in a different levels after this operation....